EXOSIMS/SurveySimulation/multiSS.py

```diff
@@ -0,0 +1,904 @@
+        [X, Y] = np.meshgrid(comps, comps)
+        c_mat = self.coeff[2] * (X + Y)
+
+        # kill diagonal with arbitrary low number
+        np.fill_diagonal(c_mat, 1e-9)
+
+        # kill the upper half because the elements
+        # are symmetrical (eg. comp(a,b), comp(b,a),
+        # completeness assumed to be constant in
+        # Time for one set of observation)
+        np.tril(c_mat)
+
+        if self.count_1 == 1:
+            if len(sInds.tolist()) > 0 and old_sInd is not None:
+                # choose sInd of next target
+
+                sInd, waitTime = self.choose_next_target(
+                    old_sInd, sInds, slewTimes[sInds], intTimes[sInds]
+                )
+                waitTime = float(waitTime.value) * u.d
+
+                # Should Choose Next Target decide there are no stars it wishes to
+                # observe at this time.
+                if sInd is None and (waitTime is not None):
+                    self.vprint(
+                        "There are no stars available to observe. Waiting {}".format(
+                            waitTime
+                        )
+                    )
+                    return DRM, None, None, waitTime
+                elif (sInd is None) and (waitTime is None):
+                    self.vprint(
+                        "There are no stars available to observe and waitTime is None."
+                    )
+                    return DRM, None, None, waitTime
+                # store selected star integration time
+                intTime = intTimes[sInd]
+
+            # if no observable target, advanceTime to next Observable Target
+            else:
+                self.vprint(
+                    "No Observable Targets at currentTimeNorm= "
+                    + str(TK.currentTimeNorm.copy())
+                )
+                return DRM, None, None, None
+
+        # populate DRM with occulter related values
+        if OS.haveOcculter and self.count_1 == 1:
+            if self.count == 0:
+                DRM = Obs.log_occulterResults(
+                    DRM, slewTimes_2[sInd], sInd, sd_2[sInd], dV[sInd]
+                )
+                self.count = self.count + 1
+            else:
+                DRM = Obs.log_occulterResults(
+                    DRM, slewTimes[sInd], sInd, sd[sInd], dV_2[sInd]
+                )
+                self.count = 0
+
+            return DRM, sInd, intTime, waitTime
+
+        extraTimes = int(np.ceil(Obs.settlingTime.to("day").copy().value)) + int(
+            np.ceil(mode["syst"]["ohTime"].copy().value)
+        )
+
+        if old_sInd is None and self.counter_2 is None:
+            i = 0
+            # change the int values to ceil to check for keepout
+            while self.ko_2 == 0:
+                # H = np.unravel_index(c_mat.argmax(), c_mat.shape)
+                # first_target = sInds[H[0]]
+                # second_target = sInds[H[1]]
+
+                # for generating a random schedule,
```

Comment:
again - can't do this.  if you want to give the random walk as an option, it has to be toggled by a user input.

Reply:
Created a separate module, fixed